homeassistant/components/geniushub/binary_sensor.py

```diff
@@ -34,6 +29,33 @@ async def async_setup_platform(
     async_add_entities(switches, update_before_add=True)
 
 
+async def async_setup_entry(
```

Comment:
Instead of creating the do setup functions, can we move all that logic in the async setup entry?